core/genesis.go

```diff
@@ -246,10 +245,7 @@ func SetupGenesisBlock(
 			return newcfg, stored, compatErr
 		}
 	}
-	// Don't overwrite if the old is identical to the new
-	if newData, _ := json.Marshal(newcfg); !bytes.Equal(storedData, newData) {
-		rawdb.WriteChainConfig(db, stored, newcfg)
-	}
```

Comment:
should we instead comment out these lines and add an extra warning why this breaks?

Reply:
Good call, adding a comment to explain the discrepancy with upstream. Not going to leave the commented out code though.